ansible/roles/ocp4-workload-rhte-analytics_data_ocp_workshop/tasks/workload.yml

```diff
@@ -198,23 +193,101 @@
     cluster_apps_domain: "{{ cluster_apps_domain_out.resources[0].spec.domain}}"
     notebook_suffix: "/notebooks/data-engineering-and-machine-learning-workshop.git/source/notebooks/hybrid-data-engineering.ipynb"
 
-- name: Insert user data into CSV file
-  lineinfile: 
-    path: /tmp/ausgeben.csv
-    line: "user{{ item }},openshift,{\"Notebook URL\":\"http://jupyterhub-opendatahub-user{{ item }}.{{ cluster_apps_domain }}/user/user{{ item }}{{ notebook_suffix }}\"}"
-  loop: "{{ range(1, num_users | int + 1, 1) | list }}"
+- name: make sure the guides project exists
+  k8s:
+    state: present
+    definition:
+      apiVersion: project.openshift.io/v1
+      kind: Project
+      metadata:
+        name: guides
+
+- name: search for username distribution tool
+  k8s_facts:
+    kind: DeploymentConfig
+    name: get-a-username
+    namespace: guides
+  register: r_gau_dc
+
+- name: deploy redis
+  when: r_gau_dc.resources | list | length == 0
+  shell: >
+    oc process -f -
+    -p DATABASE_SERVICE_NAME=redis
+    -p REDIS_PASSWORD=redis
+    -p VOLUME_CAPACITY=1Gi
+    -p MEMORY_LIMIT=1Gi
+    -p REDIS_VERSION=latest
+    | oc create -n guides  -f -
+  args:
+    stdin: "{{ lookup('file', './files/redis-template.yaml') }}"
+
+- name: wait for redis to be ready
+  when: r_gau_dc.resources | list | length == 0
+  k8s_facts:
+    api_version: v1
+    kind: Pod
+    namespace: guides
+    label_selectors:
+      - name = redis
+      - deploymentconfig = redis
+    field_selectors:
+      - status.phase=Running
+  register: r_redis_pod
+  retries: 120
+  delay: 10
+  until: r_redis_pod.resources | list | length == 1
+  
+- name: deploy username distribution tool
+  when: r_gau_dc.resources | list | length == 0
+  shell: >
+    oc -n guides new-app quay.io/openshiftlabs/username-distribution:1.2 --name=get-a-username
+    -e LAB_REDIS_HOST=redis
+    -e LAB_REDIS_PASS=redis
+    -e LAB_TITLE={{ 'Data Engineering Workshop' | quote }}
+    -e LAB_DURATION_HOURS=8h
+    -e LAB_USER_COUNT={{ num_users }}
+    -e LAB_USER_ACCESS_TOKEN={{ 'openshift' | quote }}
+    -e LAB_USER_PASS={{ 'openshift' | quote }}
+    -e LAB_USER_PREFIX=user
+    -e LAB_USER_PAD_ZERO=false
+    -e LAB_ADMIN_PASS={{ 'openshift' | quote }}
+    -e LAB_MODULE_URLS={{ ( 'http://jupyterhub-opendatahub-%USERNAME%.' + cluster_apps_domain + '/user/%USERNAME%' + notebook_suffix + ';Workshop Jupyter Notebook') | quote }}
+
+- name: expose username distribution tool
+  when: r_gau_dc.resources | list | length == 0
+  k8s:
+    api_version: route.openshift.io/v1
+    namespace: guides
+    state: present
+    kind: Route
+    definition:
+      metadata:
+        name: get-a-username
+        labels:
+          app: get-a-username
+      spec:
+        host: ''
+        to:
+          kind: Service
+          name: get-a-username
+          weight: 100
+        port:
+          targetPort: 8080-tcp
+  register: Route
```

Comment:
This registers a variable, `Route`, which is then not used. Best to name this variable `r_route`, using the `r_` prefix standard for registered variables to avoid conflicts. This variable could be used to eliminate the need to call `k8s_facts` in the next task.